### PR TITLE
Feature/embed javafx

### PR DESCRIPTION
**Pull request by Shadowrs**
NotificationsUI may not have been used, and yes V3 is JavaFX based, but I don't want to wait. 

Here is a fix for "Application launch must not be called more than once" ... Application-based javafx means javafx is the core of the program. This moves it over to swing-embedded, where it can be run any number of times.

### CODE REVIEW

File: src/main/java/org/parabot/api/ui/JavaFxUtil.java [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+    private final File       target;
+    private final Class<?> controller;
+    private final ProgressListener listener;
+
+    /**
+     * Constructor to use when stylesheet.fxml is in the Parabot Jar
+     * @param controllerPath Location to .fxml such as "/storage/ui/notifications.fxml"
+     */
+    public JavaFxUtil(final String controllerPath, final Class<?> controller) {
+        URL end1;
```
Comment (JKetelaar) [inline src/main/java/org/parabot/api/ui/JavaFxUtil.java]: Why not just call it `end`?

File: src/main/java/org/parabot/api/ui/JavaFxUtil.java [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+        try {
+            end1 = Configuration.class.getClass().getResource(controllerPath).toURI().toURL();
```
Comment (JKetelaar) [inline src/main/java/org/parabot/api/ui/JavaFxUtil.java]: `this.end = Configuration.class.getClass().getResource(controllerPath).toURI().toURL();`

File: src/main/java/org/parabot/api/ui/JavaFxUtil.java [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        } catch (MalformedURLException | URISyntaxException e) {
+            end1 = null;
```
Comment (JKetelaar) [inline src/main/java/org/parabot/api/ui/JavaFxUtil.java]: You can remove this line, default value is already null

Comment (Shadowrs) [reply src/main/java/org/parabot/api/ui/JavaFxUtil.java]: `end` is final and the try-catch makes it ugly, my bad that was auto generated

**Comment by Shadowrs** [comment]
That's better!

File: src/main/java/org/parabot/api/ui/JavaFxUtil.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+ * how the UI threadding works. To solve this, JavaFX should be used by embedded the FX content inside an FXPanel,
+ * which itself should just fill the content of a empty JFrame.
+ *
+ * @author Shadowrs
+ */
+public abstract class JavaFxUtil {
+
+    private       JFrame     frame;
+    private       JFXPanel   jfxp;
+    private       JavaFxUtil INSTANCE;
```
Comment (JKetelaar) [inline src/main/java/org/parabot/api/ui/JavaFxUtil.java]: Keep the naming conventions (`instance`)

**Pull request merged**